mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoAdapterTest.java

```diff
@@ -286,7 +286,7 @@ private static void populate(MongoCollection<Document> collection, URL resource)
   @Test public void testFilterRedundant() {
     assertModel(MODEL)
         .query(
-            "select * from zips where state > 'CA' and state < 'AZ' and state = 'OK'")
+            "select * from zips where state > 'CA' and state < 'PZ' and state = 'OK'")
```

Comment:
Could you please clarify the reason of this change?

Reply:
Yes, with my changes in ```RexSimplify``` this compound condition became available for simplification and since `OK` is not between `CA` and `AZ` whole condition was simplified to `false` . To preserve expected result as is I extended the range to include `OK` in it.

Reply:
Ok, thanks for the explanation.

Reply:
I would say that it makes more sense to update the plan (as you did in other tests) rather than updating the query.